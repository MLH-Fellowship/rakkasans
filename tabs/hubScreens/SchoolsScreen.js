import React from "react";
import { View, StyleSheet, Image } from "react-native";
import { useNavigation } from "@react-navigation/native";

import SquareButton from "../../components/SquareButton";
import Colors from "../../constants/Colors";
import RectButton from "../../components/RectButton";

export default function SchoolsScreen() {
  const navigation = useNavigation();
  return (
    <View style={styles.container}>
      <Image
        source={require("../../assets/images/Torri.png")}
        style={{ width: 30, height: 30, top: "-8%" }}
      />
      <View style={[styles.buttonPair, { paddingTop: 10 }]}>
        <SquareButton
          name="school"
          text="AASLT"
          buttonSize={80}
          textSize={13}
          iconSize={50}
          onPress={() =>
            navigation.navigate("School", {
              name: "AASLT",
              // image: require("../../assets/images/HHC.png"),
            })
          }
        />
        <SquareButton
          name="school"
          text="PRC"
          buttonSize={80}
          textSize={13}
          iconSize={50}
          onPress={() =>
            navigation.navigate("School", {
              name: "PRC",
              // image: require("../../assets/images/HHC.png"),
            })
          }
        />
      </View>
      <View style={styles.buttonPair}>
        <SquareButton
          name="school"
          text="School 3"
          buttonSize={80}
          textSize={13}
          iconSize={50}
          onPress={() =>
            navigation.navigate("School", {
              name: "AASLT",
              // image: require("../../assets/images/HHC.png"),
            })
          }
        />
        <SquareButton
          name="school"
          text="School 4"
          buttonSize={80}
          textSize={13}
          iconSize={50}
          onPress={() =>
            navigation.navigate("School", {
              name: "AASLT",
              // image: require("../../assets/images/HHC.png"),
            })
          }
        />
      </View>
      <View style={styles.buttonPair}>
        <SquareButton
          name="school"
          text="School 5"
          buttonSize={80}
          textSize={12}
          iconSize={50}
          onPress={() =>
            navigation.navigate("School", {
              name: "AASLT",
              // image: require("../../assets/images/HHC.png"),
            })
          }
        />
        <SquareButton
          name="school"
          text="School 6"
          buttonSize={80}
          textSize={13}
          iconSize={50}
          onPress={() =>
            navigation.navigate("School", {
              name: "AASLT",
              // image: require("../../assets/images/HHC.png"),
            })
          }
        />
      </View>
      <View style={[styles.buttonPair, { paddingBottom: 25 }]}>
        <SquareButton
          name="school"
          text="School 7"
          buttonSize={80}
          textSize={13}
          iconSize={50}
          onPress={() =>
            navigation.navigate("School", {
              name: "AASLT",
              // image: require("../../assets/images/HHC.png"),
            })
          }
        />
        <SquareButton
          name="school"
          text="School 8"
          buttonSize={80}
          textSize={13}
          iconSize={50}
          onPress={() =>
            navigation.navigate("School", {
              name: "AASLT",
              // image: require("../../assets/images/HHC.png"),
            })
          }
        />
      </View>
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white,
    width: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  buttonPair: {
    width: "85%",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-evenly",
    paddingBottom: 30,
  },
});
