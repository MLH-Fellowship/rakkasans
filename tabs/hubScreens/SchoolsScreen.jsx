import React, { useState, useEffect } from 'react';
import { View, StyleSheet, Image } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import axios from 'axios';
import SquareButton from '../../components/SquareButton';
import Colors from '../../constants/Colors';

export default function SchoolsScreen() {
  const navigation = useNavigation();
  const [resources, setResources] = useState([]);

  useEffect(() => {
    try {
      const getSchoolResources = async () => {
        const request = await axios.get('http://localhost:3001/resources');
        console.log(request.data);
        const schoolResources = request.data.filter(
          (data) => data.resource_type === 'school',
        );

        setResources(schoolResources);
      };

      getSchoolResources();
    } catch (error) {
      console.log(error);
    }
  }, []);
  const styles = StyleSheet.create({
    container: {
      flexWrap: 'wrap',
      alignItems: 'center',
      flexDirection: 'row',
      justifyContent: 'center',
      backgroundColor: Colors.white,
      width: '100%',
      height: '100%',
    },
    buttonView: {
      width: '45%',
      alignItems: 'center',
      justifyContent: 'space-evenly',
      paddingBottom: 30,
    },
    image: {
      marginTop: '5%',
      marginBottom: '15%',
      width: '100%',
    },
  });
  return (
    <View style={styles.container}>
      <View style={styles.image}>
        <Image
          source={require('../../assets/images/Torri.png')}
          style={{
            width: 50,
            height: 50,
            paddingTop: '10%',
            alignSelf: 'center',
          }}
        />
      </View>
      {resources.map(
        ({
          id,
          image,
          icon_name,
          resource_title,
          phone_number,
          email,
          grad_times,
          location,
          report_times,
        }) => (
          <View style={[styles.buttonView]}>
            <SquareButton
              id={id}
              name={icon_name}
              text={resource_title}
              buttonSize={65}
              textSize={13}
              iconSize={38}
              onPress={() => navigation.navigate('Resource', {
                image,
                name: resource_title,
                email,
                location,
                phone: phone_number,
                grad_times,
                report_times,
                // image: require("../../assets/images/HHC.png"),
              })}
            />
          </View>
        ),
      )}
    </View>
  );
}
