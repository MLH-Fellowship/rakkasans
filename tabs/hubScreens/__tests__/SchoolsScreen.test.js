import React from 'react';
import renderer from 'react-test-renderer';

import SchoolsScreen from '../SchoolsScreen';

jest.useFakeTimers();
jest.mock('@react-navigation/native');

it('renders correctly', () => {
  const tree = renderer.create(<SchoolsScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
