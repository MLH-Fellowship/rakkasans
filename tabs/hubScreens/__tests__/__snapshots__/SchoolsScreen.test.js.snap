// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 30,
        "top": "-8%",
        "width": 30,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "paddingBottom": 30,
          "width": "85%",
        },
        Object {
          "paddingTop": 10,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          AASLT
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          PRC
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 30,
        "width": "85%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          School 3
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          School 4
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "paddingBottom": 30,
        "width": "85%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 12,
              },
            ]
          }
        >
          School 5
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          School 6
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "paddingBottom": 30,
          "width": "85%",
        },
        Object {
          "paddingBottom": 25,
        },
      ]
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          School 7
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderRadius": 15,
          "elevation": 5,
          "opacity": 1,
          "padding": 10,
          "shadowColor": "#a0a0a0",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.4,
          "shadowRadius": 8,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 80,
              "width": 80,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#333333",
                "paddingBottom": 6,
                "textAlign": "center",
              },
              Object {
                "fontFamily": "andale-mono",
              },
              Object {
                "fontSize": 13,
              },
            ]
          }
        >
          School 8
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 25,
              "height": 50,
              "justifyContent": "center",
              "width": 50,
            }
          }
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
</View>
`;
